static/js/shared/stat_types.ts

```diff
@@ -27,77 +27,57 @@ export interface StatMetadata {
   unit?: string;
 }
 
-export interface TimeSeries {
-  val?: {
-    [date: string]: number; // Date might be "latest" if it's from the cache
-  };
-  metadata?: StatMetadata;
+export interface Observation {
+  date: string;
+  value: number;
+  facet?: string;
 }
 
-// rsponse from /api/stats
-export interface StatApiResponse {
-  [place: string]: {
-    data: {
-      [statVar: string]: TimeSeries | null;
-    };
-    name?: string;
-  };
+export interface Series {
+  // series field is guaranteed to set from Flask.
+  series: Observation[];
+  facet?: string;
 }
 
-// response from /place/displayname
-export interface DisplayNameApiResponse {
-  [placeDcid: string]: string;
-}
-
-export interface SourceSeries {
-  provenanceDomain: string;
-  val: { [key: string]: number };
-  importName?: string;
-  measurementMethod?: string;
-  observationPeriod?: string;
-  scalingFactor?: string;
-  unit?: string;
-  mprop?: string;
-}
-
-// response from /api/stats/all
-export interface StatAllApiResponse {
-  placeData: {
-    [place: string]: {
-      statVarData: {
-        [statVar: string]: {
-          sourceSeries: SourceSeries[];
-        };
-      };
+export interface SeriesApiResponse {
+  facets: Record<string, StatMetadata>;
+  data: {
+    [variable: string]: {
+      [entity: string]: Series;
     };
   };
 }
 
-export interface PlacePointStatData {
-  date: string;
-  value: number;
-  metaHash?: number;
-  metadata?: StatMetadata;
-}
-export interface PlacePointStat {
-  metaHash?: number;
-  stat: Record<string, PlacePointStatData>;
+export interface SeriesAllApiResponse {
+  facets: Record<string, StatMetadata>;
+  data: {
+    [variable: string]: {
```

Comment:
I think it'll make typing even stronger if we added an intermediate type for `Record<String, ObsSeries>`, `Record<String, Observation>`, etc. We declare these repeatedly in the utils and it'll help trace these types to the response. Wdyt? There might be too much to update at this point though.

Reply:
Added new types to represent these.